### ISSUE TITLE
Virtualenv created with pyenv-virtualenv not fully uninstalled

### ISSUE THREAD

**Issue by bcvanmeurs** [pyenv/pyenv-virtualenv#436]
Too many issues will kill our team's development velocity, drastically.
Make sure you have checked all steps below.

### Prerequisite
* [x] Make sure your problem is not listed in [the common build problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems).
* [x] Make sure no duplicated issue has already been reported in [the pyenv issues](https://github.com/pyenv/pyenv/issues). You should look for closed issues, too.
* [x] Make sure you are not asking us to help solving your specific issue.
  * GitHub issues is opened mainly for development purposes. If you want to ask someone to help solving your problem, go to some community site like [Gitter](https://gitter.im/yyuu/pyenv), [StackOverflow](https://stackoverflow.com/questions/tagged/pyenv), etc.
* [x] Make sure your problem is not derived from packaging (e.g. [Homebrew](https://brew.sh)).
  * Please refer to the package documentation for the installation issues, etc.
* [x] Make sure your problem is not derived from plugins.
  * This repository is maintaining `pyenv` and the default `python-build` plugin only. Please refrain from reporting issues of other plugins here.

### Description
- [x] Platform information (e.g. Ubuntu Linux 16.04): macOS 12.5.1
- [x] OS architecture (e.g. amd64): x86_64
- [x] pyenv version: 2.3.4
- [x] Python version: 3.9.7
- [ ] C Compiler information (e.g. gcc 7.3): 
- [x] Please attach the debug trace of the failing command as a gist:
  * Run `env PYENV_DEBUG=1 <faulty command> 2>&1 | tee trace.log` and attach `trace.log`. E.g. if you have a problem with installing Python, run `env PYENV_DEBUG=1 pyenv install -v <version> 2>&1 | tee trace.log` (note the `-v` option to `pyenv install`).
https://gist.github.com/bcvanmeurs/2ec119b39f034a12993d19017bc177cb#file-trace-log

Pyenv seems not to uninstall everything correctly:

```shell
❯ pyenv virtualenv 3.9.7 test
❯ pyenv versions
* system (set by /Users/bram/.pyenv/version)
  3.9.7
  3.10.5
  3.9.7/envs/test
  3.10.5/envs/global
* global (set by /Users/bram/.pyenv/version)
  test
❯ pyenv uninstall test
pyenv: remove /Users/bram/.pyenv/versions/test? [y|N] y
pyenv: test uninstalled
❯ pyenv versions
* system (set by /Users/bram/.pyenv/version)
  3.9.7
  3.10.5
  3.9.7/envs/test
  3.10.5/envs/global
* global (set by /Users/bram/.pyenv/version)
```

In this example the environment `test` is gone but `3.9.7/envs/test` is still there.
But this doesn't work:

```shell
❯ pyenv uninstall 3.9.7/envs/test
pyenv: version `test' not installed
```

`test` env is gone here:
```shell
❯ ls ~/.pyenv/versions
3.10.5                 3.9.7                                global
```

`test` env remains here:
```shell
❯ ls ~/.pyenv/versions/3.9.7/envs
test
```
manually removing `~/.pyenv/versions/3.9.7/envs/test` is a workaround

Could this be related to https://github.com/pyenv/pyenv/pull/2432 in 2.3.4?

**Comment by golkedj** [comment]
I just started encountering this issue today as well. Please fix, it is very annoying to have to remove the directory in `.pyenv`

**Comment by mdmould** [comment]
> I just started encountering this issue today as well. Please fix, it is very annoying to have to remove the directory in `.pyenv`

For a temporary fix, follow https://github.com/pyenv/pyenv#upgrading to roll back to v2.3.3.

**Comment by golkedj** [comment]
> > I just started encountering this issue today as well. Please fix, it is very annoying to have to remove the directory in `.pyenv`
> 
> For a temporary fix, follow https://github.com/pyenv/pyenv#upgrading to roll back to v2.3.3.

I used the automatic installer for installation. Can you please describe or link to specific instructions on how to update/roll back to a specific version?

**Comment by mdmould** [comment]
> > > I just started encountering this issue today as well. Please fix, it is very annoying to have to remove the directory in `.pyenv`
> > 
> > 
> > For a temporary fix, follow https://github.com/pyenv/pyenv#upgrading to roll back to v2.3.3.
> 
> I used the automatic installer for installation (https://github.com/pyenv/pyenv-installer). Can you please describe or link to specific instructions on how to update/roll back to a specific version?

See here: https://github.com/pyenv/pyenv-installer#install. To make a fresh install, first uninstall pyenv with `rm -rf $PYENV_ROOT` (note that this will delete all of your pyenv Python versions and virtual environments). Then set the environment variable `export PYENV_GIT_TAG=v2.3.3` and install pyenv with `curl https://pyenv.run | bash` followed by `exec $SHELL`.

**Comment by native-api** [comment]
Looks like this is fixed in https://github.com/pyenv/pyenv-virtualenv/pull/438

**Comment by golkedj** [comment]
> Looks like this is fixed in #438

It definitely is fixed for me! Thanks

**Comment by bcvanmeurs** [comment]
> Looks like this is fixed in #438

Great, thanks!

**Comment by native-api** [comment]
> Actually, even HEAD fails with:
> 
> ```
> Error: inreplace failed
> /usr/local/Cellar/pyenv-virtualenv/HEAD-c4de255/bin/pyenv-virtualenv-prefix:
>   expected replacement of "\"${BASH_SOURCE%/*}\"/../libexec" with "/usr/local/Cellar/pyenv-virtualenv/HEAD-c4de255/libexec"
> ```

okay, a release is in order then
